app/controllers/api/pages_controller.rb

```diff
@@ -1,12 +1,59 @@
+require 'rack'
+
 class Api::PagesController < ApplicationController
+  before_action :get_page
   layout false
 
   def show
-    @page = Page.find(params[:id])
-
     respond_to do |format|
       format.json
     end
   end
+
+  def update
+    updater = PageUpdater.new(@page, page_url(@page))
+    if updater.update(all_params)
+      render json: { refresh: updater.refresh?, id: @page.id }, status: :ok
+    else
+      render json: { errors: shallow_errors(updater.errors) }, status: 422
+    end
+  end
+
+  def share_rows
+    render json: (@page.shares.map do |s|
+      {html: render_to_string(partial: "share/#{s.name}s/summary_row", locals: {share: s, page: @page})}
+    end)
+  end
+
+  private
+
+  def all_params
+    # this is pretty janky but it's the best I can do moving quickly
```

Comment:
Can you write one sentence on top of this long comment explaining really shortly what this does? (Builds a nested params hash?) The long comment doesn't feel super helpful without context.